packages/berry-core/sources/Configuration.ts

```diff
@@ -81,16 +81,31 @@ export enum SettingsType {
   LOCATOR_LOOSE = 'LOCATOR_LOOSE',
   STRING = 'STRING',
   SECRET = 'SECRET',
+  OBJECT = 'OBJECT',
```

Comment:
Names are obviously open for discussion, `OBJECT` means a predefined set of properties, `MAP` means the value type defined but the keys aren't.

```yaml
npmScopes:  # <-- this is a map
  "@private": # <-- this is an object
    npmRegistryServer: "https://my.private/registry"
```
```

Reply:
I wonder whether it would make sense to use something like `prop-types` later on, if we need to model complex schemas 🤔

Reply:
Oh btw I'd vote for either `SHAPE`, `SCHEMA`, or `INTERFACE` instead of `OBJECT` - it seems clearer to me that the data is expected to have a specific set of properties (the `Object` type is traditionally a loose type in type systems).